lib/pipefy_message/consumer.rb

```diff
@@ -104,6 +104,7 @@ def process_message
                                     log_text: "Message received by consumer poller, processed " \
                                               "in #{elapsed_time_ms} milliseconds"
                                   }, context, correlation_id, event_id))
+          $stdout.flush # Ensure everything is flushed to log after processing a message
```

Comment:
**question:** Why _do_ we have this buffer, though? Just wondering if there could potentially be any side effects.

Also, should we perhaps do this in other places as well? Maybe even wrap logger calls with that?

cc @jleber

Reply:
Buffering stdout in memory increase throughput because it requires less input/output operations to file.

Reply:
I agree that it makes sense to have this here, especially at this point in time (testing) in which observability matters more than speed. Worst case scenario, it might slightly decrease performance, and then, if we ever turn this off, we get a good surprise :smile: